sound/hybrid_font.h

```diff
@@ -138,25 +137,43 @@ class HybridFont : public SaberBase {
       PlayPolyphonic(effect);
     }
   }
-  
-  void StartSwing() override {
-    if (!guess_monophonic_) {
-      if (swing_player_) {
-        // avoid overlapping swings, based on value set in ProffieOSSwingOverlap.  Value is
-        // between 0 (full overlap) and 1.0 (no overlap)
-        if (swing_player_->pos() / swing_player_->length() >= config_.ProffieOSSwingOverlap) {
-          swing_player_->set_fade_time(swing_player_->length() - swing_player_->pos());
-          swing_player_->FadeAndStop();
-          swing_player_.Free();
-          swing_player_ = PlayPolyphonic(&swng);
+
+  void StartSwing(const Vec3& gyro, float swingThreshold_, float slashThreshold_) override {
+    float speed = sqrtf(gyro.z * gyro.z + gyro.y * gyro.y);
+    if (speed > swingThreshold_) {
+      if (!guess_monophonic_) {
+        if (swing_player_) {
+          // avoid overlapping swings, based on value set in ProffieOSSwingOverlap.  Value is
+          // between 0 (full overlap) and 1.0 (no overlap)
+          if (swing_player_->pos() / swing_player_->length() >= config_.ProffieOSSwingOverlap) {
+            swing_player_->set_fade_time(swing_player_->length() - swing_player_->pos());
+            swing_player_->FadeAndStop();
+            swing_player_.Free();
+          }
         }
+        if (!swing_player_){
+          if (accel_.len() > slashThreshold_ && slsh.files_found() && !swinging_) {
```

Comment:
This doesn't make any sense to me.
accel_.len() is the length of the acceleration vector, however, without subtracting out the "down" vector, the length isn't very meaningful. Also, the way this is written, I would imagine that up and down swings would get very different results for slashes.

If we subtract out the "down" vector, accel.len() will give us how fast the hilt is accelerating in some direction.
However, hilt movement and blade movement aren't really the same thing. For some styles of swinging they might be the same, but not for all.

Reply:
In testing this value seemed consistent for triggering on a slashing motion (a long fast swing) and the values were reliably larger on the faster motions.   i can experiment with the down vector and see how things change.

Reply:
Like I pointed out before: It's not enough that it works, it must also make sense.
Code that just works, but doesn't make sense is impossible to maintain, update or rewrite.

Reply:
Ok, so tested with subtracting fusor.down from accel.  Slsh seems to be easier to trigger now (which makes sense, because the value is lower due to the above) and it seems more consistent.  I did some testing with the modified accel.len printing during swings and it seems very consistently tied to swing speed no matter how the hilt is moving.  So this change appears to make triggering a slash a little more repeatable.

Reply:
It's still weird to me to use acceleration for this though, as hilt acceleration comes from moving the hilt. But I think of a "slash" as moving the *blade* quickly, not the hilt.

Also, accel_ - fusor.down() is the same as mss, so we should probably just have a fusor.mss() function IF that is what we should use for slash detection.

Reply:
Ok, I can change to mss easily enough.  However, what do you think would be a better metric to detect blade vs hilt acceleration?

Reply:
I think you mean "what would be a better metric to detect slash vs. swing" ?

I still think the rate-of-change for the gyro might be a good way to do this. The fusor actually calculates this already, but doesn't expose it. I can try to fix that....

Reply:
I don't know if you already saw this, but try using this for the threshold:

// fusor.gyro_slope() is in degrees per second per microsecond
Vec3 gyro_slope = fusor.gyro_slope() * 1000000; 
float dss = sqrtf(gyro_slope.z * gyro_slope.z + gyro_slope.y * gyro_slope.y);

(Then use "dss" instead of accel_.len())

Reply:
I was playing with slope but didn't think to add the multiplier.  I'll do that and see what I come up with.

Reply:
Might be better to put that multiplier in the fusor, as degrees per second per second is a much more reasonable unit than degrees per second per microsecond.

Reply:
hmm, the values seem very high.  I'm seeing deg/s/s values of 20k +

Reply:
That does seem high.
Unfortunately I haven't used this code enough to develop an intuition for whether that is unreasonable or not. I also haven't tested the code very much, so there could be bugs...

I remember we tried calculating delta-gyro before, but I don't remember what range of values you ended up with then.

Reply:
The same issue occurred, where the values were crazy high in addition to being inconsistent.  It seems pretty unlikely that I could accelerate a hilt at 20 Km/s. ;)  The good news is that the values this time seem to be more consistent than before so even though the high numbers seem off, they're consistently off.

Reply:
The unit is not m/s though. It's degrees per second per second, which is very different.
If we convert everything to radians, it might be more like meters per second though.
20k degrees per second = 348 radians per second per second. If the blade is one meter, the tip would experience an acceleration of 348 m/s/s, which is about 35G, which seems high, but not inconceivable. If we had an accelerometer in the tip of the blade we could test it. :)

Reply:
Ok, did some more testing with the dss value.  It seems consistent and tunable, so success.  I've decided to have the value in radians/s/s so we don't have to have values in the config file in the 5 digit range.